### PR TITLE
Lattice60

### PR DESCRIPTION
**Pull request by emdarcher**
Adding a new keyboard called the Lattice60.

## Description

Adding a new through-hole ATmega328p driven 60% keyboard called the Lattice60.
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lattice60/config.h [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x16c0
+#define PRODUCT_ID      0x27db
```
Comment (drashna) [inline keyboards/lattice60/config.h]:
```suggestion
#define VENDOR_ID       0x16C0
#define PRODUCT_ID      0x27DB
```

File: keyboards/lattice60/config.h [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    emdarcher
+#define PRODUCT         LATTICE60
+#define DESCRIPTION     A HHKB layout keyboard using only through-hole components.
```
Comment (drashna) [inline keyboards/lattice60/config.h]: This value isn't used, IIRC.

```suggestion
```

File: keyboards/lattice60/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool led_update_user(led_t led_state) {
+    return true;
+}
+*/
```
Comment (drashna) [inline keyboards/lattice60/keymaps/default/keymap.c]: This block can be removed.
```suggestion
```

File: keyboards/lattice60/readme.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+Make example for this keyboard (after setting up your build environment):
+
+    make lattice60:default
+
+To activate the USBasploader bootloader, hold the BOOT button down during power on or reset. LED2 should light up to indicate that the bootloader is active.
+
+Flash the firmware using the `make` command with the `program` parameter, example below:
+
+    make lattice60:default:program
```
Comment (drashna) [inline keyboards/lattice60/readme.md]:
```suggestion
    make lattice60:default:flash
```

File: keyboards/lattice60/rules.mk [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+F_CPU = 12000000
+
+# disable the debugging code
+OPT_DEFS = -DDEBUG_LEVEL=0
+
+# custom command for programming via USBasp for the 
+# ATmega328p running USBasploader
+# can be flashed with the following QMK make parameters:
+# make lattice60:default:program
+PROGRAM_CMD = avrdude -c usbasp -p atmega328p -U flash:w:$(BUILD_DIR)/$(TARGET).hex
```
Comment (drashna) [inline keyboards/lattice60/rules.mk]: This shouldn't be needed.  There is the `USBasp` target, which is called by the `flash` target. 
```suggestion
```

File: keyboards/lattice60/config.h (hunk 2 above)
Comment (emdarcher) [reply keyboards/lattice60/config.h]: Even if it isn't used I can still keep it there right?

Comment (drashna) [reply keyboards/lattice60/config.h]: Yeah, that's not a problem.

File: keyboards/lattice60/lattice60.c [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    // put your per-action keyboard code here
+    // runs for every action, just before processing by the firmware
+
+    return process_record_user(keycode, record);
+}
+
+*/
```
Comment (fauxpark) [inline keyboards/lattice60/lattice60.c:65]:
```suggestion
```

File: keyboards/lattice60/keymaps/emdarcher/keymap.c [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+void matrix_scan_user(void) {
+
+}
+
+bool led_update_user(led_t led_state) {
+    return true;
+}
+*/
```
Comment (fauxpark) [inline keyboards/lattice60/keymaps/emdarcher/keymap.c:74]:
```suggestion
```

File: keyboards/lattice60/keymaps/emdarcher/keymap.c [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+    _FN0,
+    _FN1
+};
+
+// Defines the keycodes used by our macros in process_record_user
+/*
+enum custom_keycodes {
+};
+*/
+
```
Comment (fauxpark) [inline keyboards/lattice60/keymaps/emdarcher/keymap.c:30]:
```suggestion
```

File: keyboards/lattice60/lattice60.h [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    { k02,   k12,   k22,   k32,   k42,   k52,   k62,   k72 }, \
+    { k03,   k13,   k23,   k33,   k43,   k53,   k63, KC_NO }, \
+    { k04,   k14,   k24,   k34,   k44,   k54,   k64,   k74 }, \
+    { k05,   k15,   k25,   k35,   k45,   k55,   k65,   k75 }, \
+    { k06,   k16,   k26,   k36,   k46,   k56,   k66,   k76 }, \
+    { k07,   k17,   k27,   k37,   k47,   k57,   k67,   k77 }  \
+}
+
+
+#define LAYOUT_hhkb( \
```
Comment (noroadsleft) [inline keyboards/lattice60/lattice60.h:48]: This conforms to the `60_hhkb` community layout.

```suggestion
#define LAYOUT_60_hhkb( \
```

File: keyboards/lattice60/info.json [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+            ]
+        },
+        "LAYOUT_hhkb": {
```
Comment (noroadsleft) [inline keyboards/lattice60/info.json:13]:
```suggestion
        "LAYOUT_60_hhkb": {
```

File: keyboards/lattice60/keymaps/default_hhkb/keymap.c [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/lattice60/keymaps/default_hhkb/keymap.c:26]:
```suggestion
    [_BASE] = LAYOUT_60_hhkb(
```

File: keyboards/lattice60/keymaps/default_hhkb/keymap.c [hunk 12]
```diff
@@ -0,0 +1,40 @@
...
+        KC_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSLS, KC_GRV,
+        KC_TAB,   KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,    KC_BSPC,
+        KC_LCTL,   KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,     KC_ENT,
+        KC_LSFT,     KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, MO(_FN),
+                KC_LALT, KC_LGUI, /*        */ KC_SPC,                  KC_RGUI, KC_RALT
+    ),
+    [_FN] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/lattice60/keymaps/default_hhkb/keymap.c:33]:
```suggestion
    [_FN] = LAYOUT_60_hhkb(
```

File: keyboards/lattice60/keymaps/emdarcher/keymap.c [hunk 13]
```diff
@@ -0,0 +1,48 @@
...
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN0,
+    _FN1
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/lattice60/keymaps/emdarcher/keymap.c:27]:
```suggestion
    [_BASE] = LAYOUT_60_hhkb(
```

File: keyboards/lattice60/keymaps/emdarcher/keymap.c [hunk 14]
```diff
@@ -0,0 +1,48 @@
...
+        KC_GESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_BSLS, KC_GRV,
+        KC_TAB,   KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC,    KC_BSPC,
+        KC_LCTL,   KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,     KC_ENT,
+        KC_LSFT,     KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSFT, MO(_FN0),
+                KC_LALT, KC_LGUI, /*        */ KC_SPC,                  MO(_FN1), KC_RALT
+    ),
+    [_FN0] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/lattice60/keymaps/emdarcher/keymap.c:34]:
```suggestion
    [_FN0] = LAYOUT_60_hhkb(
```

File: keyboards/lattice60/keymaps/emdarcher/keymap.c [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+        KC_GRV, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_INS, KC_DEL,
+        KC_CAPS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_PSCR, KC_SLCK, KC_PAUS, KC_UP, KC_TRNS, KC_BSPC,
+        KC_TRNS,  KC_VOLD, KC_VOLU, KC_MUTE, KC_TRNS, KC_TRNS, KC_TRNS, KC_VOLU, KC_HOME, KC_PGUP, KC_LEFT, KC_RGHT,    KC_PENT,
+        KC_TRNS,    KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_MPLY, KC_VOLD, KC_END, KC_PGDN, KC_DOWN, KC_TRNS, KC_TRNS,
+                KC_TRNS, KC_TRNS,           KC_TRNS,                    KC_TRNS, KC_TRNS
+    ),
+    [_FN1] = LAYOUT_hhkb(
```
Comment (noroadsleft) [inline keyboards/lattice60/keymaps/emdarcher/keymap.c:41]:
```suggestion
    [_FN1] = LAYOUT_60_hhkb(
```

File: keyboards/lattice60/rules.mk [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (noroadsleft) [inline keyboards/lattice60/rules.mk:38]:
```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs

LAYOUTS = 60_hhkb
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**